File: day18/consecutive_sequence.py
import random

def longest_consecutive(nums):
    num_set = set(nums)
    longest = 0

    for num in num_set:
        if num - 1 not in num_set:
            current = num
            streak = 1
            while current + 1 in num_set:
                current += 1
                streak += 1
            longest = max(longest, streak)

    return longest

def main():
    length = random.randint(10, 30)
    nums = random.sample(range(0, 100), length)
    print("Input:", nums)
    print("Longest consecutive sequence length:", longest_consecutive(nums))

if __name__ == "__main__":
    main()

File: day18/consecutiveSequence.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
#include <algorithm>
#include <random>
#include <ctime>

int longestConsecutive(const std::vector<int>& nums) {
    std::unordered_set<int> numSet(nums.begin(), nums.end());
    int longest = 0;

    for (int num : numSet) {
        if (numSet.find(num - 1) == numSet.end()) {
            int current = num;
            int streak = 1;

            while (numSet.find(current + 1) != numSet.end()) {
                current++;
                streak++;
            }

            longest = std::max(longest, streak);
        }
    }

    return longest;
}

int main() {
    std::mt19937 rng(static_cast<unsigned int>(std::time(nullptr)));
    std::uniform_int_distribution<int> lengthDist(10, 30);
    std::uniform_int_distribution<int> valueDist(0, 99);

    int length = lengthDist(rng);
    std::unordered_set<int> uniqueNums;
    while (uniqueNums.size() < static_cast<size_t>(length)) {
        uniqueNums.insert(valueDist(rng));
    }

    std::vector<int> nums(uniqueNums.begin(), uniqueNums.end());

    std::cout << "Input: ";
    for (int n : nums) std::cout << n << " ";
    std::cout << "\nLongest consecutive sequence length: " << longestConsecutive(nums) << "\n";

    return 0;
}
